scripts/link_directory_metadata.py

```diff
@@ -0,0 +1,222 @@
+    """
+
+    variables = {"column_name": column_name}
+
+    response = backend._execute_query(query=query, variables=variables)
+    response = backend._simplify_graphql_response(response)["allColumn"]
+
+    df = pd.json_normalize(response)
+
+    starts_with = df["name"].str.startswith(column_name + "_")
+    ends_with = df["name"].str.endswith("_" + column_name)
+    contains = df["name"].str.contains("_" + column_name + "_")
+    is_exactly = df["name"].str == column_name
+
+    selected = starts_with | ends_with | contains | is_exactly
+
+    df = df[selected]
+
+    return df
+
+
+def get_directory_column_id(
+    directory_column_name, directory_table_name, backend
+) -> str:
+    """
+    Get the directory id from that column
+    """
+    query = """ query($column_name: String)
+        {allColumn(name: $column_name) {
+                edges {
+                node {
+                    name
+                    _id
+                    table {
+                    dataset {
+                        fullSlug
+                    }
+                    slug
+                    }
+                }
+                }
+            }
+            }
+    """
+
+    variables = {"column_name": directory_column_name}
+    response = backend._execute_query(query=query, variables=variables)
+    response = backend._simplify_graphql_response(response)["allColumn"]
+    df = pd.json_normalize(response)
+
+    colunas_de_diretorio = df["table.dataset.fullSlug"].str.contains("diretorios")
+    for index, coluna in df[colunas_de_diretorio].iterrows():
+        if coluna["table.slug"] == directory_table_name:
+            print(
+                f"\nConnecting to the directory column: \n\t{coluna['table.dataset.fullSlug']}.{coluna['table.slug']}:{coluna['name']} "
+            )
+            return coluna["_id"]
+
+    print(
+        f"\nWARNING - Unable to find the directory column with the following information: \n\tcolumn_name: {directory_column_name}  \n\ttable: {directory_table_name}"
+    )
+
+
+def modify_directory_metadata(column_id: str, directory_id: str, backend) -> None:
+    """
+    modify directory metadata
+    """
+    mutation = """
+                    mutation($input: CreateUpdateColumnInput!) {
+                        CreateUpdateColumn(input: $input) {
+                            errors {
+                                field,
+                                messages
+                            },
+                            clientMutationId,
+                            column {
+                                id,
+                            }
+                        }
+                    }
+        """
+    variables = {"id": column_id, "directoryPrimaryKey": directory_id}
+
+    headers = get_headers(backend)
+
+    backend._execute_query(
+        query=mutation, variables={"input": variables}, headers=headers
+    )
+
+    return None
+
+
+def select_columns(df, not_column_name, remove_already_done=False):
```

Comment:
Não sei se `not_column_name` é `str`
```suggestion
def select_columns(df: pd.DataFrame, not_column_name, remove_already_done: bool=False) -> list[str]:
```